### PR TITLE
DockerImagePuller installed image first

### PR DESCRIPTION
**Pull request by alexfdezsauco**
DockerImagePuller check first if the image is available locally before execute the pull command

### CODE REVIEW

**Comment by alexfdezsauco** [comment]
Sorry, this include wrong files.

File: src/Microsoft.Tye.Hosting/DockerRunner.cs [hunk 1]
```diff
@@ -164,7 +164,13 @@ async Task RunDockerContainer(IEnumerable<(int ExternalPort, int Port, int? Cont
                     }
                 }
 
-                var command = $"run -d {workingDirectory} {volumes} {environmentArguments} {portString} --name {replica} --restart=unless-stopped {docker.Image} {docker.Args ?? ""}";
+                var network = string.Empty;
+                if (!string.IsNullOrWhiteSpace(docker.Network))
```
Comment (davidfowl) [inline src/Microsoft.Tye.Hosting/DockerRunner.cs]: Undo, this came from your other PR.

File: src/Microsoft.Tye.Hosting/Model/DockerRunInfo.cs [hunk 2]
```diff
@@ -21,5 +21,7 @@ public DockerRunInfo(string image, string? args)
         public string? Args { get; }
 
         public string Image { get; }
+
+        public string Network { get; set; }
```
Comment (davidfowl) [inline src/Microsoft.Tye.Hosting/Model/DockerRunInfo.cs]: Remove

File: src/tye/ApplicationBuilderExtensions.cs [hunk 3]
```diff
@@ -50,6 +50,8 @@ public static Application ToHostingApplication(this ApplicationBuilder applicati
                         dockerRunInfo.VolumeMappings.Add(new DockerVolume(mapping.Source, mapping.Name, mapping.Target));
                     }
 
+                    dockerRunInfo.Network = container.Network;
```
Comment (davidfowl) [inline src/tye/ApplicationBuilderExtensions.cs]: Remove

**Comment by alexfdezsauco** [comment]
I created a new pull request with a single file.

**Pull request closed without merging**